geopandas/decorators.py

```diff
@@ -0,0 +1,47 @@
+from textwrap import dedent
+from typing import Callable, Union
+
+
+def doc(*docstrings: Union[str, Callable], **params) -> Callable:
```

Comment:
If it is code taken from pandas we should make a note of it in docstring to give them proper credit.